sqlx/sqlx_service.c

```diff
@@ -241,6 +241,21 @@ _configure_with_arguments(struct sqlx_service_s *ss, int argc, char **argv)
 	return TRUE;
 }
 
+static void
+_patch_configuration_maxrss(void)
+{
+	if (sqliterepo_max_rss <= 0) {
+		struct rlimit rl = {};
+		int rc = getrlimit(RLIMIT_DATA, &rl);
+		if (rc == 0) {
+			gint64 maxrss = rl.rlim_cur;
+		    if (maxrss <= 0)
+				maxrss = G_MAXINT64;
+			sqliterepo_max_rss = maxrss;
```

Comment:
we should emit a log with detected `sqliterepo_max_rss`

Reply:
Emitting logs in signal handlers is not an excellent idea (despite the fact we already do).
The glib log handler might requires locking, and the signal handler might be fired while a log was already being emitted, leading to a deadlock.